### ISSUE TITLE
`gclient sync` landmines error after upgrade to python 3.7

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#87960]
**Description:**
The failure happens when executing `gclient sync`.

**Error:**
```
________ running 'python3 src/build/landmines.py' in '/Users/aaclarke/dev/engine'
Clobbering due to:
--- old_landmines	Mon Jul 19 15:25:21 2021
+++ new_landmines	Mon Aug  9 11:25:26 2021
@@ -1 +1 @@
-Lets start a new landmines file.
+b"Let's start a new landmines file."
Traceback (most recent call last):
  File "src/build/landmines.py", line 135, in <module>
    sys.exit(main())
  File "src/build/landmines.py", line 129, in main
    clobber_if_necessary(landmines)
  File "src/build/landmines.py", line 80, in clobber_if_necessary
    clobber.clobber(out_dir)
  File "/Users/aaclarke/dev/engine/src/build/clobber.py", line 99, in clobber
    delete_build_dir(path)
  File "/Users/aaclarke/dev/engine/src/build/clobber.py", line 60, in delete_build_dir
    shutil.rmtree(build_dir)
  File "/usr/local/Cellar/python@3.7/3.7.11/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 504, in rmtree
    onerror(os.path.islink, path, sys.exc_info())
  File "/usr/local/Cellar/python@3.7/3.7.11/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 502, in rmtree
    raise OSError("Cannot call rmtree on a symbolic link")
OSError: Cannot call rmtree on a symbolic link
Error: Command 'python3 src/build/landmines.py' returned non-zero exit status 1 in /Users/aaclarke/dev/engine
Clobbering due to:
--- old_landmines	Mon Jul 19 15:25:21 2021
+++ new_landmines	Mon Aug  9 11:25:26 2021
@@ -1 +1 @@
-Lets start a new landmines file.
+b"Let's start a new landmines file."
Traceback (most recent call last):
  File "src/build/landmines.py", line 135, in <module>
    sys.exit(main())
  File "src/build/landmines.py", line 129, in main
    clobber_if_necessary(landmines)
  File "src/build/landmines.py", line 80, in clobber_if_necessary
    clobber.clobber(out_dir)
  File "/Users/aaclarke/dev/engine/src/build/clobber.py", line 99, in clobber
    delete_build_dir(path)
  File "/Users/aaclarke/dev/engine/src/build/clobber.py", line 60, in delete_build_dir
    shutil.rmtree(build_dir)
  File "/usr/local/Cellar/python@3.7/3.7.11/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 504, in rmtree
    onerror(os.path.islink, path, sys.exc_info())
  File "/usr/local/Cellar/python@3.7/3.7.11/Frameworks/Python.framework/Versions/3.7/lib/python3.7/shutil.py", line 502, in rmtree
    raise OSError("Cannot call rmtree on a symbolic link")
OSError: Cannot call rmtree on a symbolic link
```

**Platform:**
macOS 11.5.1

**Note:** 
python3.7 is required to run the luci.py script.

**Comment by gaaclarke** [comment]
ping @cbracken, https://github.com/flutter/buildroot/pull/495 is likely the cause.

**Comment by cbracken** [comment]
Taking a look -- thanks for the heads-up.

**Comment by chinmaygarde** [comment]
Can we just remove `landmines.py` and the associated `.landmines`? We don't use these and its an unnecessary gclient step.

**Comment by cbracken** [comment]
Yeah was just looking and a diff here nukes the file but we never actually write these. Sending a patch.

**Comment by cbracken** [comment]
Closed by flutter/engine#27967
